src/CacheClient.cpp

```diff
@@ -111,15 +123,23 @@ void GetCachingBehaviourSettingsFromEnvironment(bool* attemptReads, bool* attemp
     if (behaviour == nullptr)
         Croak("%s is set, but %s is not.", kENV_CACHE_SERVER_ADDRESS,kENV_BEE_CACHE_BEHAVIOUR);
 
-    if (0 == strcmp("RW", behaviour))
+    for (const char* c_ptr = behaviour; ; c_ptr++)
+    {
+        char c = *c_ptr;
+        if (c == 0)
+            break;
+        if (c == 'R')
     {
         *attemptReads = true;
-        *attemptWrites = true;
+            continue;
     }
-    if (0 == strcmp("R", behaviour))
-        *attemptReads = true;
-    if (0 == strcmp("W", behaviour))
+        if (c == 'W')
+        {
         *attemptWrites = true;
+            continue;
+        }
+        Croak("The cache behaviour string provided: %s contains a character that is not R or W", behaviour);
+    }
 
-    Log(kDebug, "Caching enabled with %s=%s %s=%s and mode: %s%s\n", kENV_CACHE_SERVER_ADDRESS, server, kENV_REAPI_CACHE_CLIENT, reapi_cache_client, *attemptReads ? "R":"_", *attemptWrites ? "W":"_");
+    Log(kDebug, "Caching enabled with %s=%s %s=%s and mode: %s%s%s\n", kENV_CACHE_SERVER_ADDRESS, server, kENV_REAPI_CACHE_CLIENT, reapi_cache_client, *attemptReads ? "R":"_", *attemptWrites ? "W":"_");
```

Comment:
Isn't this two `%s` too much? I see 7 `%s`, but only 5 arguments. Did you mean to _remove_ a `%s`, instead of adding one?